enterprise_access/apps/subsidy_access_policy/utils.py

```diff
@@ -40,3 +50,16 @@ def request_cache():
     Helper that returns a namespaced RequestCache instance.
     """
     return RequestCache(namespace=CACHE_NAMESPACE)
+
+
+def create_idempotency_key_for_transaction(subsidy_uuid, **metadata):
+    """
+    Create a key that allows a transaction to be created idempotently.
+    """
+    idpk_data = {
+        tx_key: value
+        for tx_key, value in metadata.items()
+        if tx_key in TRANSACTION_METADATA_KEYS
+    }
+    hashed_metadata = hashlib.md5(str(idpk_data).encode()).hexdigest()
+    return f'{LEDGERED_SUBSIDY_IDEMPOTENCY_KEY_PREFIX}-{subsidy_uuid}-{hashed_metadata}'
```

Comment:
Is it cool that I'm dropping the quantity from the transaction idempotency key sent by enterprise-access?  That context just doesn't exist at the point of calling the subsidy redeem API endpoint from access, and logically I don't think it's needed for de-duplicating requests anyway